pkg/log/log.go

```diff
@@ -64,6 +66,9 @@ func init() {
 		loggerWasSetLock.Lock()
 		defer loggerWasSetLock.Unlock()
 		if !loggerWasSet {
+			// Alert the user that this is not a good thing to rely on
+			fmt.Fprintln(os.Stderr, "warning: log.SetLogger was not called; this is not recommended and may be unsupported in a future version of controller-runtime.  Call log.SetLogger(log.NullLogger{}) if you want no log output.")
```

Comment:
Ci failure is genuine, maybe add a lint exception?
```
pkg/log/log.go:70: line is 223 characters (lll)
```

Ref https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_controller-runtime/1446/pull-controller-runtime-test-master/1375495983600766976#1:build-log.txt%3A214

Reply:
Hi, buddies. It seems something like log buffer is need, cause If the program is directly exiting within 30 seconds, we neither got stderr as line 70 showed nor other logs anyhow.

@alvaroaleman Hi, buddy, can I have a try for this?